project/cmake/weblib/emcc.txt

```diff
@@ -1,5 +1,10 @@
 set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -s DISABLE_EXCEPTION_CATCHING=1")
 
+option(sizeprofile "Generate debuginfo for release build" OFF)
+if(sizeprofile)
+    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Wno-limited-postlink-optimizations -g -gseparate-dwarf=cvizzu.debug.wasm -gsource-map")
+endif()
+
```

Comment:
I think this is not the proper way of handle the debug infos.
There is a RelWithDebInfo CMAKE build target, and that can be used.

```patch
Subject: [PATCH] RELWITHDEBINFO
---
Index: project/cmake/emcc.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/project/cmake/emcc.txt b/project/cmake/emcc.txt
--- a/project/cmake/emcc.txt	(revision 751e7a705d5bd140a0d3b2954a3ef9cf13b3e030)
+++ b/project/cmake/emcc.txt	(date 1690362848530)
@@ -5,7 +5,15 @@
 
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Oz -flto")
 
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -gsource-map")
+set(DEBUG_CXX_FLAGS "-g -gsource-map")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${DEBUG_CXX_FLAGS} \
+    -gseparate-dwarf=cvizzu.debug.wasm")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${DEBUG_CXX_FLAGS}")
 
-set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} -Wno-limited-postlink-optimizations \
-    -gsource-map --source-map-base http://127.0.0.1:8080/example/lib/")
+set(DEBUG_LINKER_FLAGS "-Wno-limited-postlink-optimizations \
+    -gsource-map")
+
+set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} \
+    ${DEBUG_LINKER_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG} \
+    ${DEBUG_LINKER_FLAGS} --source-map-base http://127.0.0.1:8080/example/lib/")
```